source/MaterialXGenShader/Util.cpp

```diff
@@ -13,119 +13,156 @@ namespace MaterialX
 namespace
 {
     const float EPS_ZERO = 0.00001f;
-    const float EPS_ONE  = 1.0f - EPS_ZERO;
-
-    bool isZero(float v)
-    {
-        return v < EPS_ZERO;
-    }
-
-    bool isZero(const Color3& c)
+    const float ONE_VALUE = 1.0f;
+    const float ZERO_VALUE = 0.0f;
+    
+    bool isEqual(const float& v, const float& v2)
     {
-        return isZero(c[0]) && isZero(c[1]) && isZero(c[2]);
+        return std::abs(v - v2) < EPS_ZERO;
     }
 
-    bool isZero(ValuePtr value)
+    bool isEqual(ValuePtr value, const float& value2)
     {
-        if (value->isA<float>() && isZero(value->asA<float>()))
+        if (value->isA<float>() && isEqual(value->asA<float>(), value2))
         {
             return true;
         }
-        if (value->isA<Color3>() && isZero(value->asA<Color3>()))
+        else if (value->isA<Color3>())
         {
-            return true;
+            const Color3& value3 = value->asA<Color3>();
+            if (isEqual(value3[0], value2) && isEqual(value3[1], value2) && isEqual(value3[2], value2))
+            {
+                return true;
+            }
         }
         return false;
     }
 
-    bool isOne(float v)
-    {
-        return v > EPS_ONE;
-    }
+    using OpaqueTestPair = std::pair<string, float>;
+    using OpaqueTestPairList = vector<OpaqueTestPair>;
 
-    bool isOne(const Color3& c)
+    // Get corresponding input for an interfacename for a nodegraph. 
+    // The check is done for any corresponding nodedef first and then for 
+    // any direct child input of the nodegraph.
+    InputPtr getInputInterface(const string& interfaceName , NodePtr node)
     {
-        return isOne(c[0]) && isOne(c[1]) && isOne(c[2]);
+        InputPtr interfaceInput = nullptr;
+        ElementPtr parent = node->getParent();
+        NodeGraphPtr nodeGraph = parent ? parent->asA<NodeGraph>() : nullptr;
+        if (nodeGraph)
+        {
+            NodeDefPtr nodeDef = nodeGraph->getNodeDef();
```

Comment:
Fix to check nodedef's input value/connection first as this overrides any default value on the functional graph's inputs.